src/vellum/workflows/workflows/tests/test_base_workflow.py

```diff
@@ -384,8 +389,57 @@ class TestWorkflow(BaseWorkflow[Inputs, State]):
     assert isinstance(state.meta.workflow_inputs, Inputs)
     assert state.meta.workflow_inputs.baz == "My input baz"
 
-    # AND the node execution cache should deserialize
-    assert state.meta.node_execution_cache
+    # AND the node execution cache should deserialize correctly
+    assert state.meta.node_execution_cache._node_executions_initiated == {NodeA: {UUID(last_span_id)}}
+    assert state.meta.node_execution_cache._node_executions_fulfilled == {NodeA: Stack.from_list([UUID(last_span_id)])}
+    assert state.meta.node_execution_cache._node_executions_queued == {NodeA: []}
+    assert state.meta.node_execution_cache._dependencies_invoked == {UUID(last_span_id): {NodeA}}
+
+
+def test_base_workflow__deserialize_legacy_node_execution_cache():
+
+    # GIVEN a state definition
+    class State(BaseState):
+        bar: str
+
+    # AND a node
+    class NodeA(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            foo: str
+
+    # AND a workflow that uses both
+    class TestWorkflow(BaseWorkflow[BaseInputs, State]):
+        graph = NodeA
+
+    # WHEN we deserialize the state that had a legacy node execution cache format
+    last_span_id = str(uuid4())
+    node_a_full_name = "vellum.workflows.workflows.tests.test_base_workflow.test_base_workflow__deserialize_legacy_node_execution_cache.<locals>.NodeA"  # noqa: E501
```

Comment:
since we have records in clickhouse with this old way, we will forever maintain both. This also leaves he door open in the future if we want to relitigate absolute module paths as identifiers instead of ids